File: sdk/typescript/src/architecture.ts
/**
 * TypeScript SDK Client for Enterprise AI Architecture Platform (Phase 5.26).
 */

export class ArchitectureClient {
  private client: any;

  constructor(client: any) {
    this.client = client;
  }

  async createNode(name: string, nodeType: string, environment: string = 'production', attributes: Record<string, any> = {}): Promise<any> {
    return this.client.request('POST', '/v1/architecture/nodes', {
      name,
      node_type: nodeType,
      environment,
      attributes,
    });
  }

  async listNodes(environment?: string): Promise<any[]> {
    return this.client.request('GET', '/v1/architecture/nodes', undefined, { environment });
  }

  async addDependency(sourceNodeId: string, targetNodeId: string, dependencyType: string = 'DEPENDS_ON'): Promise<any> {
    return this.client.request('POST', '/v1/architecture/dependencies', {
      source_node_id: sourceNodeId,
      target_node_id: targetNodeId,
      dependency_type: dependencyType,
    });
  }

  async getTopology(environment: string = 'production'): Promise<any> {
    return this.client.request('GET', '/v1/architecture/topology', undefined, { environment });
  }

  async createSnapshot(environment: string = 'production', description: string = 'Snapshot'): Promise<any> {
    return this.client.request('POST', '/v1/architecture/snapshots', { environment, description });
  }

  async proposeChange(idempotencyKey: string, actionType: string, targetNodeIds: string[]): Promise<any> {
    return this.client.request('POST', '/v1/architecture/changes', {
      idempotency_key: idempotencyKey,
      action_type: actionType,
      target_node_ids: targetNodeIds,
    });
  }

  async getDrift(): Promise<any[]> {
    return this.client.request('GET', '/v1/architecture/drift');
  }

  async getTrust(): Promise<any> {
    return this.client.request('GET', '/v1/architecture/trust');
  }

  async getAnalytics(): Promise<any> {
    return this.client.request('GET', '/v1/architecture/analytics');
  }
}

File: sdk/go/architecture.go
package llmengine

import (
	"context"
	"net/http"
)

type ArchitectureClient struct {
	client *Client
}

func NewArchitectureClient(c *Client) *ArchitectureClient {
	return &ArchitectureClient{client: c}
}

type CreateNodeRequest struct {
	Name        string                 `json:"name"`
	NodeType    string                 `json:"node_type"`
	Environment string                 `json:"environment,omitempty"`
	Attributes  map[string]interface{} `json:"attributes,omitempty"`
}

type ArchitectureNode struct {
	NodeID     string                 `json:"node_id"`
	TenantID   string                 `json:"tenant_id"`
	Name       string                 `json:"name"`
	NodeType   string                 `json:"node_type"`
	Status     string                 `json:"status"`
	Attributes map[string]interface{} `json:"attributes,omitempty"`
}

func (a *ArchitectureClient) CreateNode(ctx context.Context, req CreateNodeRequest) (*ArchitectureNode, error) {
	var resp ArchitectureNode
	err := a.client.do(ctx, http.MethodPost, "/v1/architecture/nodes", req, &resp)
	if err != nil {
		return nil, err
	}
	return &resp, nil
}

func (a *ArchitectureClient) GetTopology(ctx context.Context, env string) (map[string]interface{}, error) {
	var resp map[string]interface{}
	err := a.client.do(ctx, http.MethodGet, "/v1/architecture/topology?environment="+env, nil, &resp)
	if err != nil {
		return nil, err
	}
	return resp, nil
}
